packages/itwinui-react/src/core/Tile/TileNew.tsx

```diff
@@ -0,0 +1,657 @@
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'button', TileTypeIndicatorOwnProps>;
+TileTypeIndicator.displayName = 'Tile.TypeIndicator';
+
+// ----------------------------------------------------------------------------
+// Tile.Badge component
+type TileBadgeOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const TileBadge = polymorphic.div('iui-tile-thumbnail-badge-container');
+TileBadge.displayName = 'Tile.Badge';
+
+// ----------------------------------------------------------------------------
+// Tile.Name component
+type TileNameOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const TileName = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', children, ...rest } = props;
+  const {
+    status,
+    isLoading,
+    isSelected,
+    isNew,
+    isActionable,
+    isDisabled,
+    onClick,
+  } = useSafeContext(TileContext);
+
+  return (
+    <Element className='iui-tile-name' ref={ref} {...rest}>
+      <TitleIcon
+        isLoading={isLoading}
+        isSelected={isSelected}
+        isNew={isNew}
+        status={status}
+      />
+      <span className='iui-tile-name-label'>
+        {isActionable && onClick ? (
+          <LinkAction
+            as='button'
+            onClick={!isDisabled ? onClick : undefined}
+            aria-disabled={isDisabled}
+          >
+            {children}
+          </LinkAction>
+        ) : (
+          children
+        )}
+      </span>
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileNameOwnProps>;
+TileName.displayName = 'TileNew.Name';
+
+// ----------------------------------------------------------------------------
+// Tile.ContentArea component
+type TileContentAreaOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const TileContentArea = polymorphic.div('iui-tile-content');
+TileContentArea.displayName = 'TileNew.ContentArea';
+
+// ----------------------------------------------------------------------------
+// Tile.Description component
+type TileDescriptionOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const TileDescription = polymorphic.div('iui-tile-description');
+TileDescription.displayName = 'TileNew.Description';
+
+// ----------------------------------------------------------------------------
+// Tile.Metadata component
+type TileMetadataOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const TileMetadata = polymorphic.div('iui-tile-metadata');
+TileMetadata.displayName = 'TileNew.Metadata';
+
+// ----------------------------------------------------------------------------
+// Tile.MoreOptions component
+type TileMoreOptionsOwnProps = {
+  children: React.ReactNode[];
+};
+
+const TileMoreOptions = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', className, children, ...rest } = props;
+  const [isMenuVisible, setIsMenuVisible] = React.useState(false);
+
+  return (
+    <DropdownMenu
```

Comment:
Real use case we are missing: We need to allow users define their own DropdownMenu for moreOptions (At least expose onClick which expands menu).

Reply:
do you mean the button that opens the menu?

Reply:
Yes, one team wants to load custom options on moreOptions button click so they need onClick event exposed.

Reply:
I added an `onClick` prop, is that what you mean?